lib/rules/no-dupe-class-members.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+/**
+ * Code inspired form: https://github.com/eslint/eslint/blob/2c7431d6b32063f74e3837ee727f26af215eada7/lib/rules/no-dupe-class-members.js
+ * Author: Toru Nagashima
+ * Copyright JS Foundation and other contributors, https://js.foundation
+ */
+module.exports = {
+    meta: {
+        docs: {
+            description: 'disallow duplicate class members',
+            category: 'LWC',
+            url: docUrl('no-dupe-class-members'),
+        },
+        schema: [],
+        messages: {
+            unexpected: "Duplicate name '{{name}}'.",
+        },
+    },
+
+    create(context) {
+        let stack = [];
+
+        /**
+         * Gets state of a given member name.
+         * @param {string} name - A name of a member.
+         * @param {boolean} isStatic - A flag which specifies that is a static member.
+         * @returns {Object} A state of a given member name.
+         *   - retv.init {boolean} A flag which shows the name is declared as normal member.
+         *   - retv.get {boolean} A flag which shows the name is declared as getter.
+         *   - retv.set {boolean} A flag which shows the name is declared as setter.
+         */
+        function getState(name, isStatic) {
+            const stateMap = stack[stack.length - 1];
```

Comment:
do we need a guard here since the stack is initialized to empty or is ClassBody visitor guarantees the existence?

Reply:
No need to verify here, I would rather fail loud and clear if the `stack` is empty which mean that the code is in an unrecoverable state at this point.